### ISSUE TITLE
Oblivion script API: ChildGroup() ignores overrides of worldspace Tamriel

### ISSUE THREAD

**Issue by LenaWolf** [TES5Edit/TES5Edit#1360]
API function ChildGroup() does not work correctly at least on the main worldspace in Oblivion - Tamriel (I didn't test on other worldspaces). It returns the number of elements defined in Oblivion.esm completely ignoring the extensions to Tamriel made in ESPs.

My concrete example: I am using Onra's TWMP, so the Tamriel worldspace is greatly extended in Tamriel.esp. Yet the ChildGroup() function insists on returning the origina world group that only contains the cells defined in Oblivion.esm, completely ignoring the extension.

The problem with this is when I try to access any cell outside the vanilla ones, it cannot be found. For example, we have a set of scripts for working with worldspaces, but they cannot look beyond the vanilla space. I have been trying to move an area in the North of Skyrim for example, using the "Worldspace region move.pas" script, and that script relies on the ChildGroup() function to cycle through the cells in the worldspace. Well, this doesn't work because it never gets to cycle through the entire worldspace.

If you need more information, you can contact me on Nexus Mods, my user name is LenaWolfBravil.

Thank you.

**Comment by zilav** [comment]
Works fine here, applied it to WRLD override in TES4Edit and it properly returns WRLD children group from overriding plugin
```
unit userscript;

function Process(e: IInterface): Integer;
begin
  if Signature(e) = 'WRLD' then
    AddMessage(FullPath(ChildGroup(e)));
end;

end.

**Comment by LenaWolf** [comment]
I just tested with your latest version 4.1.5f, and it still doesn't work. In fact, the script I am referring to, does not even compile (line 31: double should be integer). Once I got it to compile, it fails because the cells extending Tamriel cannot be found.

I can only speculate that Tamriel might be a special worldspace since this is the main worldspace in Oblivion. I use the scripts supplied with TES4Edit, my knowledge of your API is insufficient to write my own scripts. However, I compared this script with "Find cell in worldspace.pas" and the difference is that in "Find cell in worldspace" there is a loop to look in each loaded ESP separately. When I inserted that loop into the "Worldspace region move" script, it did find the cells.

Perhaps I should add that these extra cells are defined in Tamriel.esp and have the mod index of their form IDs set to the load index of that ESP and not to 00, while the worldspace Tamriel has its mod index 00 as it is defined in Oblivion.esm. 

I added a message just after line 51 with ChildGroup() to print the number of elements found, and it returns 17 instead of hundreds of elements that should have been found if Tamriel.esp had been used. Line added:

`AddMessage(Format('Tamriel element count is %d',[ElementCount(wrldgrup)]));`

I would like to attach my modified script here, but it doesn't seem possible since this editor only accepts pictures.

**Comment by ElminsterAU** [comment]
It's clear to me that you haven't got the slightest clue what you are doing.

ChildGroup works exactly as it should. You are using it wrong. 

It will, as zilav demonstrated, return the ChildGroup of exactly the record on which you called it. 

You seem to expect it to magically collect information across multiple files. There is no such thing.

**Comment by LenaWolf** [comment]
You are right, I am stumbling in the dark. Where can I find documentation of your API, please? All I have so far is one line and it isn't enough.

I didn't write those Worldspace scripts - they come with xEdit.

**Comment by zilav** [comment]
Almost all worldspace scripts were written by request from Beyond Skyrim, Skywind, Skyblivion, TTW and other projects involving original big worldspaces (not modifications of vanilla ones). They assume you have a single plugin with entire worldspaces inside of it and don't take any overrides into consideration. In your case you have to iterate and process overrides yourself obviously.

https://tes5edit.github.io/docs/13-Scripting-Functions.html

**Comment by LenaWolf** [comment]
Thank you for the link. It wasn't obvious in any way that Worldspace scripts came with those assumptions. I have already figured out something like that through dissecting the code. It won't work for me at all. :( It does all sorts of wrong things.

It could be helpful to have such assumptions listed somewhere. I would not have wasted days and days trying to get it to work on several separate occasions otherwise. I would have certainly not bothered debugging it so that it would even compile.

**Comment by zilav** [comment]
All scripts are provided as is, they basically serve as examples on how to do different things via scripts in xEdit. What they do (or not) is clear by looking at the actual code. Take any suitable demo script and modify to your needs, or write you own one from scratch. Or just find someone to write for you.
